actionview/lib/action_view/lookup_context.rb

```diff
@@ -280,7 +280,15 @@ def initialize_details(target, details)
     # add :html as fallback to :js.
     def formats=(values)
       if values
+        values = values.compact
         values.concat(default_formats) if values.delete "*/*"
+        values.uniq!
+
+        invalid_types = (values - Template::Types.symbols)
+        unless invalid_types.empty?
```

Comment:
I think `if invalid_types.any?` would be clearer than the double negative.

Reply:
Could also rename the variable to `invalid_values` there's already values, formats, Types and symbols as language in play and the types language are on the Template.

Reply:
I agree it looks better, but ex. `[nil, false].any?` would be false, so I'd like to keep the `empty?` for being slightly more correct (though I agree more awkward).

👍 I'll fix the variable name.